v2/yarpctransport/handler_invoker.go

```diff
@@ -51,9 +51,7 @@ type StreamInvokeRequest struct {
 
 // InvokeUnaryHandler calls the handler h, recovering panics and timeout errors,
 // converting them to YARPC errors. All other errors are passed through.
-func InvokeUnaryHandler(
-	i UnaryInvokeRequest,
-) (res *yarpc.Response, buf *yarpc.Buffer, err error) {
+func InvokeUnaryHandler(i UnaryInvokeRequest) (res *yarpc.Response, buf *yarpc.Buffer, err error) {
```

Comment:
I think there’s an opportunity to reduce some package name noise at the call site by making this a Do method on UnaryInvokeRequest, maybe pivoting context into unaryInvokeRequest.Do(ctx).

Reply:
added to notes